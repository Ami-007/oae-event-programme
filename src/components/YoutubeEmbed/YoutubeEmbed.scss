.youtube-embed {
  width: 100%;
  height: 16rem;
}
