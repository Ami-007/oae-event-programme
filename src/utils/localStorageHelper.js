export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
