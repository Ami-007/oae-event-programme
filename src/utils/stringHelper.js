export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
