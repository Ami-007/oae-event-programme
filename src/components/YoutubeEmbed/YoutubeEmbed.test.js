import YoutubeEmbed from "./YoutubeEmbed";
import { customRender } from "../../utils/testUtils";
import { screen } from "@testing-library/react";

it("should render the YouTube embed component", () => {
  const { container } = customRender(<YoutubeEmbed embedId="q6EoRBvdVPQ" />);
  expect(container).toMatchSnapshot();
});

it("shouldn't render the YouTube embed component if no embedId has been input", () => {
  const { container } = customRender(<YoutubeEmbed />);
  expect(container).toMatchSnapshot();
});
