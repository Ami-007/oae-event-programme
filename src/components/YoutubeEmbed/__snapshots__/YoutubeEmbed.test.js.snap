// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the YouTube embed component 1`] = `
<div>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen=""
    class="youtube-embed"
    frameborder="0"
    src="https://www.youtube.com/embed/q6EoRBvdVPQ"
    title="Embedded youtube"
  />
</div>
`;

exports[`shouldn't render the YouTube embed component if no embedId has been input 1`] = `<div />`;
